test/spec/EditorCommandHandlers-test.js

```diff
@@ -466,9 +489,314 @@ define(function (require, exports, module) {
+
+                    var lines = defaultContent.split("\n");
+                    lines[1] = "    //function bar() {";
+                    lines[3] = "        //a();";
+
+                    var expectedText = lines.join("\n");
+
+                    testToggleLine(expectedText, [{start: { line: 1, ch: 0 }, end: { line: 1, ch: 0 }, reversed: false, primary: false },
+                                                  {start: { line: 3, ch: 0 }, end: { line: 3, ch: 8 }, reversed: false, primary: true }]);
+                });
+
+                it("should only handle each line once, but preserve primary/reversed flags on ignored selections", function () {
+
                     myEditor.setSelections([{start: {line: 1, ch: 4}, end: {line: 1, ch: 4}},
+                                        {start: {line: 1, ch: 6}, end: {line: 2, ch: 4}, primary: true},
+                                        {start: {line: 3, ch: 4}, end: {line: 3, ch: 4}},
+                                        {start: {line: 3, ch: 6}, end: {line: 3, ch: 8}, reversed: true}]);
+
+                    var lines = defaultContent.split("\n");
+                    lines[1]= "    //function bar() {";
+                    lines[2]= "    //    ";
+                    lines[3]= "        //a();";
+
+
+                    var expectedText = lines.join("\n");
+
+                    testToggleLine(expectedText, [{start: {line: 1, ch: 6}, end: {line: 1, ch: 6}, reversed: false, primary: false},
+                                                  {start: {line: 1, ch: 8}, end: {line: 2, ch: 6}, reversed: false, primary: true},
+                                                  {start: {line: 3, ch: 4}, end: {line: 3, ch: 4}, reversed: false, primary: false},
+                                                  {start: {line: 3, ch: 6}, end: {line: 3, ch: 10}, reversed: true, primary: false}]);
+                });
+
+                it("should properly toggle when some selections are already commented but others aren't", function () {
+
+                    var lines = defaultContent.split("\n");
+                    lines[1] = "    //function bar() {";
+                    lines[3] = "    a();";
+                    lines[5] = "    //}";
+                    var startingContent = lines.join("\n");
+                    myDocument.setText(startingContent);
+
+                    myEditor.setSelections([{start: {line: 1, ch: 4}, end: {line: 1, ch: 4}},
+                                            {start: {line: 3, ch: 4}, end: {line: 3, ch: 4}},
+                                            {start: {line: 5, ch: 4}, end: {line: 5, ch: 4}}]);
+
+                    lines[1] = "    function bar() {";
+                    lines[3] = "    //a();";
+                    lines[5] = "    }";
+                    var expectedText = lines.join("\n");
+
+                    testToggleLine(expectedText, [{ start : { line : 1, ch : 4 }, end : { line : 1, ch : 4 }, reversed : false, primary : false },
```

Comment:
Stylenit: the test object has different spacing than the other `testToggleLine`'s (no space before, space after `,` and `;`